src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -77,7 +82,7 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
                 });
                 credentialsTreeItem.commandArgs = [this.parent];
                 return [credentialsTreeItem];
-            } else if (this.parent.azureName && parsedError.errorType === firewallNotConfiguredErrorType) {
+            } else if (this.parent.azureName && (parsedError.errorType === firewallNotConfiguredErrorType || 'ETIMEDOUT') && !(await this.isFirewallRuleSet(this.parent))) {
```

Comment:
I think you still need this logic in `checkAuthentication`. It likely handles the same error for slightly different scenarios